pkg/trtl/config/config.go

```diff
@@ -36,12 +37,19 @@ type DatabaseConfig struct {
 }
 
 type ReplicaConfig struct {
-	Enabled        bool          `split_words:"true" default:"true"`
-	PID            uint64        `split_words:"true" required:"false"`
-	Region         string        `split_words:"true" required:"false"`
-	Name           string        `split_words:"true" required:"false"`
-	GossipInterval time.Duration `split_words:"true" default:"1m"`
-	GossipSigma    time.Duration `split_words:"true" default:"5s"`
+	Enabled        bool          `split_words:"true" default:"true" json:"enabled"`
+	PID            uint64        `split_words:"true" required:"false" json:"pid"`
+	Region         string        `split_words:"true" required:"false" json:"region"`
+	Name           string        `split_words:"true" required:"false" json:"name"`
+	GossipInterval time.Duration `split_words:"true" default:"1m" json:"gossip_interval"`
+	GossipSigma    time.Duration `split_words:"true" default:"5s" json:"gossip_sigma"`
+}
+
+type ReplicaStrategyConfig struct {
+	HostnamePID bool   `split_words:"true" default:"false"` // Set to true to use HostnamePID
+	Hostname    string `envconfig:"TRTL_REPLICA_HOSTNAME"`  // configure the hostname from the environment
+	FilePID     string `split_words:"true"`                 // Set to the PID filename path to use FilePID
+	JSONConfig  string `split_words:"true"`                 // Set to the config map path to use JSONConfig
 }
```

Comment:
I don't like this, but it's the quickest way to define a strategy. In the future, I hope we can have "strategy names" e.g. "k8s", "metal", etc. that automatically configure different approaches.